lib/portage/sync/modules/git/git.py

```diff
@@ -212,7 +212,19 @@ def verify_head(self, revision='-1'):
 				writemsg_level("!!! Verifying against specified key requires gemato-11.0+ installed\n",
 					level=logging.ERROR, noiselevel=-1)
 				return False
-			openpgp_env = gemato.openpgp.OpenPGPEnvironment()
+
+			proxy = None
+			# Set global proxy setting to Gemato ephemeral env
+			if 'http_proxy' in os.environ:
+				proxy = os.environ['http_proxy']
+
+			# Override global proxy setting with one provided in emerge configuration
+			running_config_settings = \
+				self.options['emerge_config'].running_config.settings
+			if 'http_proxy' in running_config_settings:
+				proxy = running_config_settings['http_proxy']
+
+			openpgp_env = gemato.openpgp.OpenPGPEnvironment(proxy=proxy)
```

Comment:
@mgorny: it's duplicated code amount the three classes. Normally that should be good?

Reply:
It's not that bad but maybe a helper method in base class could help.